lightning/src/ln/channelmanager.rs

```diff
@@ -628,10 +634,21 @@ pub struct ChannelDetails {
 	/// Note that there are some corner cases not fully handled here, so the actual available
 	/// inbound capacity may be slightly higher than this.
 	pub inbound_capacity_msat: u64,
+	/// True if the channel was initiated (and thus funded) by us.
+	pub is_outbound: bool,
+	/// True if the channel is confirmed, funding_locked messages have been exchanged, and the
+	/// channel is not currently being shut down. `funding_locked` message exchange implies the
+	/// required confirmation count has been reached (and we were connected to the peer at some
+	/// point after the funding transaction received enough confirmations).
+	pub is_confirmed: bool,
```

Comment:
Maybe call it `is_usable` if it matches exactly Channel method. "Confirmed" is a bit ambiguous for a) ongoing shutdown implies the channel has been confirmed and b) zero-conf channel don't expect any confirmation and are ready-to-go at creation ?

Reply:
`is_usable` is overloaded in a few places in our code already, and is a bit vague. We could add verbosity (like `is_blockchain_confirmed` or `reached_target_blockchain_confirmations`) or we could maybe refer to "locked" eg `is_blockchain_locked`. What do you think?

Reply:
Maybe `is_funding_locked` would be sufficient as it covers having enough confirmations and having exchanged the `funding_locked` messages.

That said, I don't understand the argument against `is_usable`. It is only used in `Channel` as far as I can tell and is defined in the same manner as here. Where else is it used?

Reply:
Ah, I see there is `ChannelManager::list_usable_channels` which is checking against `is_live`. Perhaps that needs to be renamed instead. Or maybe `is_live` should really be `is_usable`? Then we'd need a new name for `Channel::is_usable` to something like `is_funding_locked` or maybe simply `is_funded`.

Reply:
Maybe we rename `list_usable_channels` to `list_live_channels` or so? I like `live`/`usable` including "we're connected to the peer, in addition to `is_funding_locked` (which I'll rename this to).

Reply:
Yeah, but we'll need to be logically consistent in terminology. Currently, the description for `Channel::is_usable` seems to make sense to me, but the docs for `Channel::is_live` says it's true if the "channel is currently available for use". This seems to imply that a channel that is "usable" but not "live" can't be used, which is a contradiction.

"Live" and "usable" seem to be synonymous from a terminology perspective. The latter seems more descriptive with regards to functionality.

The current use of `is_usable` is "funding locked and not known to be closing". Does `is_funding_locked` sufficiently capture that? If not, perhaps we need a better term for "usable" given the above.

Reply:
Hmm, I actually feel like "live" is more descriptive? Live, to me, is all about us being actively connected to the peer, ie the peer (and thus channel) are "live". What about renaming `list_usable_channels` to `list_live_channels` and leaving this as `is_funding_locked` (which is really just about the state of the funding transaction).

Reply:
More descriptive in terms of functionality. Is something "live" also "usable"? Only if "live" includes "funding locked" which I don't think is obvious from the word "live" alone -- you need to read the docs to figure that out.

Regardless of which, my other point regarding `Channel::is_usable` still stands if something that is live is suppose to also be usable.

Reply:
Hmm, I guess I felt like "live" did capture that. I'm happy to go with "usable" if you feel strongly. Note that `Channel` isn't a public API, so we can rename it at will whenever :).

Reply:
You can be "actively connected to the peer" (i.e., "live") but the channel may not usable though, right? i.e., if the funding isn't locked.

Currently, the problem that I see is "live" is defined as a superset of "usable", but it seems like the opposite relation is true in actuality (i.e., when funding isn't locked). But really "live" has to do with the connection and "usable" has to do with the channel. So, maybe it should be `is_live` && `is_funding_locked` == `is_usable`?

Anyhow, it's not that I feel strongly on naming it `list_live_channels`, I'm just pointing out that there is inconsistency and ambiguity in the terminology that we are using. Some of the ambiguity is cleared up by the documentation (i.e. defining the meaning of "live") but is also contradictory (i.e., something "live" is usable but something just "usable" is not).

Reply:
I don't think we currently have any code to indicate "connected to peer but channel is not yet confirmed". We could add that too, I suppose, but for now I agree we should pick love or usable and stick to it.